xarray/core/missing.py

```diff
@@ -669,113 +693,70 @@ def interp(var, indexes_coords, method: InterpOptions, **kwargs):
-        new_x_arginds = [
-            [_x, [ndim + index for index in range(_x.ndim)]] for _x in new_x
-        ]
-        new_x_arginds = [item for pair in new_x_arginds for item in pair]
-
-        args = (var, range(ndim), *x_arginds, *new_x_arginds)
-
-        _, rechunked = chunkmanager.unify_chunks(*args)
-
-        args = tuple(
-            elem for pair in zip(rechunked, args[1::2], strict=True) for elem in pair
-        )
-
-        new_x = rechunked[1 + (len(rechunked) - 1) // 2 :]
+    in_coords, out_coords = zip(*(v for v in indexes_coords.values()), strict=True)
+    broadcast_out_coords = broadcast_variables(*out_coords)
+    out_dims = broadcast_out_coords[0].dims
 
-        new_x0_chunks = new_x[0].chunks
-        new_x0_shape = new_x[0].shape
-        new_x0_chunks_is_not_none = new_x0_chunks is not None
-        new_axes = {
-            ndim + i: new_x0_chunks[i] if new_x0_chunks_is_not_none else new_x0_shape[i]
-            for i in range(new_x[0].ndim)
-        }
+    input_core_dims = []
+    all_in_core_dims = set(indexes_coords)
+    output_broadcast_dims = set(
+        itertools.chain(dim for dim in out_dims if dim not in all_in_core_dims)
+    )
 
-        # if useful, reuse localize for each chunk of new_x
-        localize = (method in ["linear", "nearest"]) and new_x0_chunks_is_not_none
+    # scipy.interpolate.interp1d always forces to float.
+    dtype = float if not issubclass(var.dtype.type, np.inexact) else var.dtype
 
-        # scipy.interpolate.interp1d always forces to float.
-        # Use the same check for blockwise as well:
-        if not issubclass(var.dtype.type, np.inexact):
-            dtype = float
-        else:
-            dtype = var.dtype
-
-        meta = var._meta
-
-        return chunkmanager.blockwise(
-            _chunked_aware_interpnd,
-            out_ind,
-            *args,
-            interp_func=func,
-            interp_kwargs=kwargs,
-            localize=localize,
-            concatenate=True,
-            dtype=dtype,
-            new_axes=new_axes,
-            meta=meta,
-            align_arrays=False,
-        )
-
-    return _interpnd(var, x, new_x, func, kwargs)
+    output_core_dims = tuple(d for d in out_dims if d not in output_broadcast_dims)
+    input_core_dims = (
+        [tuple(indexes_coords)]
+        + [tuple(d for d in _.dims if d in all_in_core_dims) for _ in in_coords]
+        + [output_core_dims] * len(out_coords)
+    )
+    output_sizes = {k: broadcast_out_coords[0].sizes[k] for k in output_core_dims}
+    result = apply_ufunc(
+        _interpnd,
+        var,
+        *in_coords,
+        *broadcast_out_coords,
+        input_core_dims=input_core_dims,
+        output_core_dims=[output_core_dims],
+        exclude_dims=all_in_core_dims,
+        dask="parallelized",
+        kwargs=dict(interp_func=func, interp_kwargs=kwargs),
+        dask_gufunc_kwargs=dict(output_sizes=output_sizes, allow_rechunk=True),
```

Comment:
`allow_rechunk=True` matches the current behaviour where we rechunk along *all* core dimensions to a single chunk.